src/CMakeLists.txt

```diff
@@ -1,6 +1,21 @@
-# Copyright (c) 2023 The Bitcoin Core developers
+# Copyright (c) 2023-present The Bitcoin Core developers
 # Distributed under the MIT software license, see the accompanying
-# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+# file COPYING or https://opensource.org/license/mit/.
+
+#=============================
+# secp256k1 subtree
+#=============================
+set(SECP256K1_DISABLE_SHARED ON CACHE BOOL "" FORCE)
+set(SECP256K1_ENABLE_MODULE_ECDH OFF CACHE BOOL "" FORCE)
+set(SECP256K1_ENABLE_MODULE_RECOVERY ON CACHE BOOL "" FORCE)
+set(SECP256K1_ECMULT_GEN_KB 86 CACHE BOOL "" FORCE)
+include(GetTargetInterface)
+get_target_interface(core_c_flags "" core_base_interface COMPILE_OPTIONS)
+set(SECP256K1_LATE_CFLAGS ${core_c_flags} CACHE STRING "" FORCE)
```

Comment:
What are the flags added on various platforms?

Reply:
> The core_c_flags variable holds the COMPILE_OPTIONS property of the core_base_interface library, which contains:
> There are no compile options that are specific to other platforms.

Why are we adding that flag specifically, into cflags, for mingw, and no other similar flags on different platforms?

Reply:
re offline discussion: why do we have a `core_c_flags` variable, if we aren't otherwise loading or using a c compiler (and how are things ending up in it).

Reply:
- And these are cxx_flags, so they shouldn't apply to libsecp256k1? Or is it just that `try_append_cxx_flags` is a bit of a misleading name (in the end it's just COMPILE_OPTIONS?) I'm not sure after looking at https://github.com/hebasto/bitcoin/blob/cmake-staging/cmake/module/TryAppendCXXFlags.cmake .

 - Anyway, I wonder if this flag should be part of the `core_base_interface`. Just because it's useful for core, doesn't mean we should pass it also when building the dependencies.

Reply:
> re offline discussion: why do we have a `core_c_flags` variable, if we aren't otherwise loading or using a c compiler (and how are things ending up in it).

Not just if this answers your question, but the variable is defined only in the `get_target_interface(core_c_flags ...`  It's just a local variable in this file and unset further below.

Reply:
Thanks! Reworked.